src/pypromice/postprocess/csv2bufr.py

```diff
@@ -187,46 +187,34 @@ def setAWSvariables(
     setBUFRvalue(ibufr, 'hour', timestamp.hour)
     setBUFRvalue(ibufr, 'minute', timestamp.minute)
 
-    vars_dict = {
-        'relativeHumidity': 'rh_i', # DMI wants non-corrected rh
-        'airTemperature': 't_i',
-        'pressure': 'p_i',
-        'windDirection': 'wdir_i',
-        'windSpeed': 'wspd_i'
-    }
-    for bufr_key, source_var in vars_dict.items():
-        if (stid in vars_to_skip) and (source_var in vars_to_skip[stid]):
-            print('----> Skipping var: {} {}'.format(stid,source_var))
-        else:
-            setBUFRvalue(ibufr, bufr_key, row[source_var])
+    setBUFRvalue(ibufr, "relativeHumidity", row["relativeHumidity"])
+    setBUFRvalue(ibufr, "airTemperature", row["airTemperature"])
+    setBUFRvalue(ibufr, "pressure", row["pressure"])
+    setBUFRvalue(ibufr, "windDirection", row["windDirection"])
+    setBUFRvalue(ibufr, "windSpeed", row["windSpeed"])
+
+    setBUFRvalue(ibufr, 'latitude', row['latitude'])
 
     # Set position metadata
-    setBUFRvalue(ibufr, 'latitude', row['gps_lat_fit'])
-    setBUFRvalue(ibufr, 'longitude', row['gps_lon_fit'])
-    setBUFRvalue(ibufr, 'heightOfStationGroundAboveMeanSeaLevel', row['gps_alt_fit']) # also height and heightOfStation?
+    setBUFRvalue(ibufr, 'latitude', row['latitude'])
+    setBUFRvalue(ibufr, 'longitude', row['longitude'])
+    setBUFRvalue(ibufr, 'heightOfStationGroundAboveMeanSeaLevel', row['heightOfStationGroundAboveMeanSeaLevel']) # also height and heightOfStation?
 
     # The ## in the codes_set() indicate the position in the BUFR for the parameter.
     # e.g. #10#timePeriod will assign to the 10th occurence of "timePeriod", which corresponds
     # to the wind speed section. Note that both the "synopMobil" and "synopLand" templates
     # appear to have the same positions for all parameters that are set here.
     # View the output BUFR to see section keys with 'bufr_dump filename.bufr'.
-    if math.isnan(row['wspd_i']) is False:
+    if math.isnan(row['windSpeed']) is False:
         #Set time significance (2=temporally averaged)
         codes_set(ibufr, '#1#timeSignificance', 2)
         #Set monitoring time period (-10=10 minutes)
         codes_set(ibufr, '#10#timePeriod', -10)
 
     #Set measurement heights
-    if math.isnan(row['z_boom_u_smooth']) is False:
-        codes_set(ibufr,
```

Comment:
why are you dropping the use of `codes_set`?

Reply:
@ladsmund regarding switching to `setBUFRvalue` instead of `code_set`. Yes, I see that you might as well just use `setBUFRvalue` as for the other vars. Not sure why `code_set` was used directly here.

You may want to keep the `if math.isnan` checks here before calling `setBUFRvalue`. I'm not sure if the addition/subtraction when passing the args will choke if there is a NaN. Overall (whether checking for NaNs here, or in `setBUFRvalue`), it looks like the previous convention was to never set a value in the BUFR with NaN, but instead to just skip it and not set that field if it is NaN.

I cannot recall right now why this was done, or if the BUFR can handle setting Nan (and it properly casts to None or Null, etc). I think this convention of not setting any NaN values was in place before I inherited the code. The original version was written by Penny/Ken, so perhaps @PennyHow can inform further.

You could run a single station through, try it one way and then the other, and then use `bufr_dump` to compare the output?